### PR TITLE
Added New Refactored getGuardSetExpression

### PR DESCRIPTION
**Pull request by BrookFeleke**
## Description
I have added a new refactored version of the `getGuardSetExp` named `getGuardSetExpression` that uses `superpose` to increase performance and run in parallel. The new function is 
I haven't removed the old function since it is used in the module `zero-constraint-subsumption.metta`.
## Motivation and Context

- reduces the overhead of using car-atom and cdr-atom for iteration over the atom one at a time.
- have better performance if done in parallel.
- It closes issue #132
## How Has This Been Tested?

- all previous test scripts pass
- additional tests have been added to cover the new function added
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Yagth — CHANGES_REQUESTED** [review]
@BrookFeleke why can't you refactor the usage of `getGuardSetExp` in the `zero constraint subsumption` too like you did in the `delete Inconsistent handle`? Then remove the `getGuardSetExp` and change the name of your function to that?

**Comment by BrookFeleke** [comment]
> @BrookFeleke why can't you refactor the usage of `getGuardSetExp` in the `zero constraint subsumption` too like you did in the `delete Inconsistent handle`? Then remove the `getGuardSetExp` and change the name of your function to that?

I have made the requested changes and removed the old `getGuardSetExp`.

**Review by Yagth — APPROVED** [review]

File: reduct/boolean-reduct/rte-helpers.metta [hunk 1]
```diff
@@ -1,43 +1,41 @@
...
-)
+ ;; Partial function that is used with the getGuardSetExp
+ ;; It checks if an element in an expression can be a guardset or not
+ ;; example (NOT A) is guard set and will be returned
+ ;; example A is guard set and will be returned
+ ;; example (AND A B) is not guard set and will not be returned
+( = (isGuard $exp) (let*
+        (
+            ($type (get-metatype $exp))
+        ) (if (== $type Symbol)
```
Comment (Nahom32) [inline reduct/boolean-reduct/rte-helpers.metta:9]: @BrookFeleke you could've used a let expression.

File: reduct/boolean-reduct/rte-helpers.metta [hunk 2]
```diff
@@ -1,43 +1,41 @@
...
-        )
-)
-)
-)
+ ;; Partial function that is used with the getGuardSetExp
+ ;; It checks if an element in an expression can be a guardset or not
+ ;; example (NOT A) is guard set and will be returned
+ ;; example A is guard set and will be returned
+ ;; example (AND A B) is not guard set and will not be returned
+( = (isGuard $exp) (let*
```
Comment (Nahom32) [inline reduct/boolean-reduct/rte-helpers.metta:6]: @BrookFeleke the naming is confusing please can you change it? It looks like you are returning a boolean but returning a list

**Comment by Yagth** [comment]
Hey @BrookFeleke, just merged another PR and your code isn't working all of a sudden. I believe it a simple conflict that can be resolved on your side.

File: reduct/boolean-reduct/rte-helpers.metta (hunk 1 above)
Comment (BrookFeleke) [reply reduct/boolean-reduct/rte-helpers.metta]: I have updated it to use `let`.

File: reduct/boolean-reduct/rte-helpers.metta (hunk 2 above)
Comment (BrookFeleke) [reply reduct/boolean-reduct/rte-helpers.metta]: @Nahom32 I have changed it to `ifGuard`, I think that is better cause it implies that it returns it if it is a valid guard. I have added comments to explain what it returns.

**Comment by BrookFeleke** [comment]
> Hey @BrookFeleke, just merged another PR and your code isn't working all of a sudden. I believe it a simple conflict that can be resolved on your side.

Yes it was because the new `isConsistentExp` was using the older version of `getGuardExp` with 3 arguments, i have changed that and it passes all tests now.

**Review by Nahom32 — APPROVED** [review]

**Pull request merged**